src/io/leopard/toLeopard.ts

```diff
@@ -8,10 +8,71 @@ import * as prettier from "prettier";
 import Target from "../../Target";
 import { List, Variable } from "../../Data";
 
-function uniqueNameGenerator(usedNames: string[] = []) {
+// Words which are invalid for any JavaScript identifier to be, when it isn't
+// on a namespace (like `this` or `this.vars`).
+//
+// This list may be more comprehensive than it needs to be in every case,
+// erring to avoid potential issues.
+const JS_RESERVED_WORDS = [
+  "arguments",
+  "await",
+  "break",
+  "case",
+  "catch",
+  "class",
+  "const",
+  "continue",
+  "debugger",
+  "default",
+  "delete",
+  "do",
+  "else",
+  "enum",
+  "eval",
+  "export",
+  "extends",
+  "false",
+  "finally",
+  "for",
+  "function",
+  "if",
+  "implements",
+  "import",
+  "in",
+  "instanceof",
+  "interface",
+  "let",
+  "new",
+  "null",
+  "package",
+  "private",
+  "protected",
+  "public",
+  "return",
+  "static",
+  "super",
+  "switch",
+  "this",
+  "throw",
+  "true",
+  "try",
+  "typeof",
+  "yield",
+  "var",
+  "void",
+  "while",
+  "with"
+];
+
+type InputShape = "any" | "index" | "number" | "string" | "boolean" | "stack";
```

Comment:
It might be worth using a TypeScript `enum` here

Reply:
Good point.